File: apps/cargo-tracker/src/contexts/handling/domain/model/handling-voyage-number.ts
import { HandlingValidationError } from './handling-validation-error.js';

/**
 * 荷役作業に紐づく航海番号（Handling Context 固有型）。
 * Routing の VoyageNumber とは共有せず、ACL 変換の境界とする（domain-model の VoyageNumber 分離設計）。
 */
export class HandlingVoyageNumber {
  private constructor(readonly value: string) {}

  static of(raw: string): HandlingVoyageNumber {
    const normalized = raw.trim();
    if (normalized.length === 0 || normalized.length > 20) {
      throw new HandlingValidationError(`不正な航海番号: ${raw}`);
    }
    return new HandlingVoyageNumber(normalized);
  }

  equals(other: HandlingVoyageNumber): boolean {
    return this.value === other.value;
  }
}

File: apps/cargo-tracker/src/main/java/com/example/cargotracker/handling/domain/model/HandlingVoyageNumber.java
package com.example.cargotracker.handling.domain.model;

/**
 * 荷役作業に紐づく航海番号（Handling モジュール固有の型）。
 *
 * <p>Routing の {@code VoyageNumber} を参照しない（{@code domain-model.md}
 * 「VoyageNumber のコンテキスト分離設計」）。
 *
 * @param value 航海番号
 */
public record HandlingVoyageNumber(String value) {

    public HandlingVoyageNumber {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("航海番号は必須です");
        }
        value = value.strip();
    }
}
